### ISSUE TITLE
Default option to automatically select best video/audio for playback based on video being played.

### ISSUE THREAD

**Issue by wags1** [yuliskov/SmartTubeNext#626]
There needs to be a "default" option for both video and audio quality so that whatever clip you play plays back at the best (highest) video and audio quality based on the clip automatically, just like "default" in the SmartYouTubeTV app. Having to manually set the video/audio quality for every video played seems to be a step backwards in functionality from what was available in the SmartYouTubeTV app.

**Comment by foreveryoungforty** [comment]
Hello, I use Video preset in the general settings for Video player in the app to obtain that goal.
I have a android Sony KD-65XF9005 4K HDR TV, so I use  preset 4K 60fps vp9 as preset.
From that moment the app will always use the best possible quality depending on that preset and of course the source quality of the video itself.
PS. I could use even 4K 60 fps vp9+hdr, but that can give some stutter in 4K-source videos (see known pinned issue 259).
Hope this helps you ; it's somehow alternative in comparison with SmartYouTubeTv's default, but more fit to the capabilities of each different device and individual choice/taste.

**Comment by wags1** [comment]
So perhaps that is the way it already works? I'll test and report back when I have some time. What about the audio as there is no preset in settings for that? I think if just setting the preset for the max your equipment will handle does the same thing as "default" in the SmartYouTubeTV app it maybe needs to be explained that way somewhere. It would be hard to infer that just looking at the settings.

**Comment by foreveryoungforty** [comment]
Be assured , it works, I tested it all the way :-) But of course always depending on the source quality of each video on YouTube (it does not upgrade as some Bluray players do).
For audio it already chooses the best, that leaves only a choice between mp4a and opus of the highest Mbit , see HQ in the player UI of a video, but you hardly will hear the difference, even on a high-end HiFi-audio system... Once you choose opus or mp4a, it will choose that for other videos also, again if availbale in the source quality of the video itself.

**Comment by wags1** [comment]
That is good to hear. I'll test as well of course. In this case I think the only thing to change is some sort of explanation somewhere. As I mentioned before, just looking at the settings you would certainly not be able to figure out that this is the way it currently works.

**Comment by foreveryoungforty** [comment]
It is briefly explained in the FAQ-link, 4th topic. https://t.me/s/STNFAQENG

**Comment by wags1** [comment]
Yeah, I read that when I was looking for an answer to this question. It definitely did not explain it sufficiently.

**Comment by wags1** [comment]
Ran a few quick tests and this looks like it works. I think some clarification or explanation of how the settings work should be included. If I was unclear lots of folks will be unclear. I will close this issue out. Thanks.

**Comment by wags1** [comment]
Ran a few quick tests and this looks like it works. I think some clarification or explanation of how the settings work should be included. If I was unclear lots of folks will be unclear. I will close this issue out. Thanks.

**Comment by wags1** [comment]
Sorry but I'm going to have to make a significant correction. After additional testing I have found that setting the preset to the best setting your devices are capable of does NOT automatically play videos with the best setting of the video being played up to the preset. It is not the same as setting the video and audio to "Default (Optimal)" in the SmartYouTubeTV app. I tested it on a History Channel episode (S2, E9) and using SmartYouTubeTV app with both video and audio set to "Default (Optimal)" it correctly picks the highest video and audio setting. Using SmartTubeNext it does not, it picks the second best video and audio setting. This seems to be a step  backwards in functionality.

**Comment by foreveryoungforty** [comment]
The only issue I got and forgot to tell you is with live channels, it does not take the best quality (Skynews, Euronews,...), but for the channels I use, especially music videos, the quality is perfect over here.

**Comment by wags1** [comment]
> The only issue I got and forgot to tell you is with live channels, it does not take the best quality (Skynews, Euronews,...), but for the channels I use, especially music videos, the quality is perfect over here.

Unfortunately that is not the case here as per the example I provided, which is not a live channel.

**Comment by ACR-Jeff** [comment]
Closing this issue, Based on a Feature that's already available in current latest versions.
